### PR TITLE
Add secure cookie flag [closes #34]

### LINKED ISSUE #34: Lacking httpOnly and secure cookie flags?

### PR DESCRIPTION
**Pull request by nimishsinghal**
Add secure cookie flag

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -146,11 +146,12 @@ Description of each property:
...
   * This function has the following parameters:
     * `msg`: Message you want to log with the help of this function.
-* `setCookie(name, value, [, expiry])`: A function that creates the cookie. Provide a custom cookie setting implementation. Use-cases of this are implementation of cookie versioning or using the common cookie setting logic in your application. You will have to use a custom implementation of [`getCookie`]() as well.
+* `setCookie(name, value, [, expiry, secure])`: A function that creates the cookie. Provide a custom cookie setting implementation. Use-cases of this are implementation of cookie versioning or using the common cookie setting logic in your application. You will have to use a custom implementation of [`getCookie`]() as well.
   * This function has the following parameters:
     * `name`: Name of the cookie.
     * `value`: Value of the cookie.
-    * `expiry`: Expiry time (in days) of the cookie.
+    * `expiry`: Expiry time (in days) of the cookie. Default: 365 days (optional).
+    * `secure`: Either true or false, indicating if the cookie transmission requires a secure protocol (https). Default: true (optional).
```
Comment (dash1291) [inline README.md]: `Default: false`

File: README.md [hunk 2]
```diff
@@ -146,11 +146,12 @@ Description of each property:
...
   * This function has the following parameters:
     * `msg`: Message you want to log with the help of this function.
-* `setCookie(name, value, [, expiry])`: A function that creates the cookie. Provide a custom cookie setting implementation. Use-cases of this are implementation of cookie versioning or using the common cookie setting logic in your application. You will have to use a custom implementation of [`getCookie`]() as well.
+* `setCookie(name, value, [, expiry, secure])`: A function that creates the cookie. Provide a custom cookie setting implementation. Use-cases of this are implementation of cookie versioning or using the common cookie setting logic in your application. You will have to use a custom implementation of [`getCookie`]() as well.
   * This function has the following parameters:
     * `name`: Name of the cookie.
     * `value`: Value of the cookie.
-    * `expiry`: Expiry time (in days) of the cookie.
+    * `expiry`: Expiry time (in days) of the cookie. Default: 365 days (optional).
```
Comment (vaidik) [inline README.md]: `(optional)` should be added next to the name of the param, like so:

```
* `expiry` (optional): Expiry...
```

File: src/cookieApi.js [hunk 3]
```diff
@@ -2,8 +2,8 @@ import jsCookie from 'js-cookie';
 
 const getCookie = name => jsCookie.get(name);
 
-const setCookie = (name, value, expiry = 365) => {
-    jsCookie.set(name, value, { expires: expiry, path: '/' });
+const setCookie = (name, value, expiry = 365, secure = false) => {
+    jsCookie.set(name, value, { expires: expiry, path: '/', secure});
```
Comment (vaidik) [inline src/cookieApi.js]: space after `secure` for consistency

File: test/cookieApi.spec.js [hunk 4]
```diff
@@ -17,3 +17,11 @@ describe('setCookie', () => {
         expect(jsCookie.get('key2')).toEqual('val2');
     });
 });
+
+describe('setCookie', () => {
+    it('should set cookie with expiry', () => {
+        setCookie('key2', 'val2', 400);
+
+        expect(jsCookie.get('key2')).toEqual('val2');
```
Comment (vaidik) [inline test/cookieApi.spec.js]: We aren't really checking if the expiry was set correctly. I think a better thing to do here would be to spy on `jsCookie.set` and see if it was called with the right set of arguments.

And, we should do the same for testing the new `secure` param as well.

**Comment by vaidik** [comment]
Approved

**Pull request merged**